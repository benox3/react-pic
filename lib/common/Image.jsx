import React, { Component, PropTypes } from 'react';

export default class Image extends Component {
  static propTypes = {
    imgStyle: PropTypes.objectOf(
      PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
    ).isRequired,
    isBlurred: PropTypes.bool.isRequired,
    blurAmount: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]).isRequired,
  };

  getStyle() {
    const { imgStyle, isBlurred, blurAmount } = this.props;
    console.log(isBlurred, blurAmount);
    if (isBlurred) {
      return {
        ...imgStyle,
        filter: `blur(${blurAmount})`,
      };
    }

    return imgStyle;
  }

  render() {
    const { alt, image } = this.props;
    return (
      <img
        alt={alt}
        style={this.getStyle()}
        src={image.url}
      />
    );
  }
}
