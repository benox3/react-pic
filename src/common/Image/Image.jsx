import React, { Component, PropTypes } from 'react';

export default class Image extends Component {
  static propTypes = {
    alt: PropTypes.string.isRequired,
    image: PropTypes.shape({
      url: PropTypes.string.isRequired,
      width: PropTypes.number.isRequired,
    }).isRequired,
    imgStyle: PropTypes.objectOf(
      PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
    ).isRequired,
    isBlurred: PropTypes.bool.isRequired,
    blurAmount: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]).isRequired,
  };

  getStyle() {
    const { imgStyle, isBlurred, blurAmount } = this.props;
    if (isBlurred) {
      return {
        ...imgStyle,
        filter: `blur(${blurAmount})`,
      };
    }

    return imgStyle;
  }

  render() {
    const { alt, image } = this.props;
    return (
      <img
        alt={alt}
        style={this.getStyle()}
        src={image.url}
      />
    );
  }
}
