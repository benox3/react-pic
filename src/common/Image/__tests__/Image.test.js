import React from 'react';

import { mount } from 'enzyme';

import Image from '../Image';

describe('Image', () => {
  const image = { url: 'http://placehold.it/290?text=â™¥', width: 290 };
  const imgStyle = { margin: '0 auto', maxWidth: '100%', width: '100%' };

  describe('rendering', () => {
    it('should render', () => {
        expect(mount(
          <Image 
            image={image} 
            alt="" 
            imgStyle={imgStyle} 
            isBlurred 
            blurAmount={30} 
          />
        ).html()).to.not.equal(null);
    });

    it('should render a standard img element', () => {
      expect(mount(
          <Image 
            image={image} 
            alt="" 
            imgStyle={imgStyle} 
            isBlurred 
            blurAmount={30} 
          />
        ).find('img').html()).to.not.equal(null);
    });
  });

  describe('snapshots', () => {
    it('should match standard snapshot', () => {
      expect(mount(
        <Image 
          image={image} 
          alt="" 
          imgStyle={imgStyle} 
          isBlurred 
          blurAmount={30} 
        />
      )).toMatchSnapshot();
    });
  });
});
