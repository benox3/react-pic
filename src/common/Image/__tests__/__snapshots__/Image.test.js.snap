// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image snapshots should match standard snapshot 1`] = `
<Image
  alt=""
  blurAmount={30}
  image={
    Object {
      "url": "http://placehold.it/290?text=♥",
      "width": 290,
    }
  }
  imgStyle={
    Object {
      "margin": "0 auto",
      "maxWidth": "100%",
      "width": "100%",
    }
  }
  isBlurred={true}
>
  <img
    alt=""
    src="http://placehold.it/290?text=♥"
    style={
      Object {
        "filter": "blur(30)",
        "margin": "0 auto",
        "maxWidth": "100%",
        "width": "100%",
      }
    }
  />
</Image>
`;
